build.gradle.kts

```diff
@@ -106,30 +112,54 @@ subprojects {
       }
     }
   }
+}
 
-  // Configure publishing
-  pluginManager.withPlugin("com.vanniktech.maven.publish") {
-    // Configure automatic-module-name, but only for published modules
-    @Suppress("UnstableApiUsage")
-    val automaticModuleName = providers.gradleProperty("AUTOMATIC_MODULE_NAME")
-      .forUseAtConfigurationTime()
-    if (automaticModuleName.isPresent) {
-      val name = automaticModuleName.get()
-      tasks.withType<Jar>().configureEach {
-        manifest {
-          attributes("Automatic-Module-Name" to name)
+allprojects {
+  tasks.withType<DokkaTask>().configureEach {
+    dokkaSourceSets.configureEach {
+      reportUndocumented.set(false)
+      skipDeprecated.set(true)
+      jdkVersion.set(8)
+      perPackageOption {
+        matchingRegex.set("com\\.squareup.moshi\\.internal.*")
+        suppress.set(true)
+      }
+    }
+    if (name == "dokkaHtml") {
+      outputDirectory.set(rootDir.resolve("docs/1.x"))
+      dokkaSourceSets.configureEach {
+        skipDeprecated.set(true)
+        externalDocumentationLink {
+          url.set(URL("https://square.github.io/okio/2.x/okio/"))
         }
       }
     }
+  }
 
-    if (name != "codegen" && pluginManager.hasPlugin("org.jetbrains.kotlin.jvm")) {
-      apply(plugin = "org.jetbrains.dokka")
-      tasks.named<DokkaTask>("dokkaHtml") {
-        outputDirectory.set(rootDir.resolve("docs/1.x"))
-        dokkaSourceSets.configureEach {
-          skipDeprecated.set(true)
-          externalDocumentationLink {
-            url.set(URL("https://square.github.io/okio/2.x/okio/"))
+  plugins.withId("com.vanniktech.maven.publish.base") {
+    configure<MavenPublishBaseExtension> {
+      publishToMavenCentral(SonatypeHost.DEFAULT)
+      signAllPublications()
+      pom {
```

Comment:
FYI there's a helper function for loading these from the same properties as before if ya want

Reply:
Things I hate from most to hate least:

 * Gradle Groovy builds. Many ways to do each thing. Mistakes fail silently. Not discoverable.
 * Gradle properties files. One way to do each thing. Fails silently. Not discoverable.
 * Gradle Kotlin builds. IntelliJ tells me when I make a mistake. Discoverable.

So configuring stuff without the safety net of a type checker? Not unless I have no other choice.